conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,252 @@
+    io:println(t); // @output error("{ballerina}TypeCastError",message="incompatible types: 'string' cannot be cast to 'int'")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a list-constructor-expr.
+Labels: trap-expr, error-type, int, string, union-type, table-constructor-expr, multiplicative-expr
+
+function init() {
+    (table<record {|readonly string name; int id;|}> key(name))|error tbl1 = trap table key(name) [
+        {name: "John Doe", id: 9223372036854775807*2}
+    ];
+    io:println(tbl1); // @output error("{ballerina}NumberOverflow",message="int range overflow")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a object-constructor-expr.
+Labels: trap-expr, error-type, object-constructor-expr, method-call-expr, panic
+
+function init() {
+    object { function bar(); } obj = object {
+                                        error? i = ();
+                                        function bar() {
+                                            self.i = self.foo();
+                                        }
+                                        function foo() {
+                                            panic error("msg");
+                                        }
+                                    };
+    error? e = trap obj.bar();
+    io:println(e); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a new-expr.
+Labels: trap-expr, error-type, new-expr, method-call-expr, panic, error-constructor-expr
+
+class Obj {
+    function bar() {
+        panic error("msg");
+    }
+}
+
+function init() {
+    Obj obj = new();
+    error? t = trap obj.bar();
+    io:println(t); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a member-access-expr.
+Labels: trap-expr, error-type, int, array-type, member-access-expr
+
+function init() {
+    int[] i = [];
+    io:println(trap i[3]); // @output error("{ballerina/lang.array}IndexOutOfRange",message="array index out of range: index: 3, size: 0")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a function-call-expr.
+Labels: trap-expr, error-type, int, function-call-expr, array-type, union-type, error-constructor-expr
+
+function foo() returns int {
+    int[] i = [];
+    return i[3];
+}
+
+function bar() {
+    panic error("msg");
+}
+
+function init() {
+    int|error t1 = trap foo();
+    io:println(t1); // @output error("{ballerina/lang.array}IndexOutOfRange",message="array index out of range: index: 3, size: 0")
+    error? t2 = trap bar();
+    io:println(t2); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a type-cast-expr.
+Labels: trap-expr, error-type, int, string, function-call-expr, union-type, type-cast-expr
+
+function foo() returns int|string {
+    return "str";
+}
+
+function init() {
+    int|error t1 = trap <int>foo();
```

Comment:
Shall we add a test with valid type cast also?
```ballerina
int|error t2 = trap <string>foo();
```

Reply:
Added